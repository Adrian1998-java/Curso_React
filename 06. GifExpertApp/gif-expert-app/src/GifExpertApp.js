import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {


  const [categories, setCategories] = useState(["One punch", 'Samurai X', 'Dragon Ball']);

//   const handleAdd = (value) => {
//       setCategories( [value, ...categories] );
//   }



    return (
    <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr/>
        <ol>
            { 
                categories.map( (category, i) => {
                    return <li key={ category }> {category} </li>
                } ) 
            }
        </ol>
    </>
  )
}
