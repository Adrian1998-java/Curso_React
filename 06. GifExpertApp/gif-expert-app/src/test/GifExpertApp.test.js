import { shallow } from "enzyme"
import React from "react"
import { GifExpertApp } from "../GifExpertApp"

describe('Pruebas sobre GifExpertApp', () => { 
    
    const wrapper = shallow( <GifExpertApp /> );

    test('Debe de funcionar correctamente', () => { 

        expect( wrapper ).toMatchSnapshot();
     });

     test('Debe demostrar una lista de categorias', () => { 
         
        const categorias = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categorias}/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categorias.length );


      })
 })