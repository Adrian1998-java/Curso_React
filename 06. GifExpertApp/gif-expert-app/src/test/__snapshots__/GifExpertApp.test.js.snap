// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp Debe de funcionar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre GifExpertApp Debe demostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
